VocaDbWeb.Core/Code/WebApi/RequireSslAttribute.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace VocaDb.Web.Code.WebApi
+{
+	[AttributeUsage(AttributeTargets.Method | AttributeTargets.Class, AllowMultiple = false)]
+	public class RequireSslAttribute : ActionFilterAttribute
+	{
+		// Code from: https://stackoverflow.com/questions/31617345/what-is-the-asp-net-core-mvc-equivalent-to-request-requesturi/40721652#40721652
+		private static bool IsSSL(HttpRequest request) => request is not null && new Uri(request.GetDisplayUrl()) is Uri requestUri && requestUri.Scheme == Uri.UriSchemeHttps;
+
+		public override void OnActionExecuting(ActionExecutingContext context)
+		{
+#if DEBUG
+			return;
+#endif
+
+			if (!IsSSL(context.HttpContext.Request))
+			{
+				// 403.4 - SSL required.
+				context.Result = new ForbidResult("This API requires SSL");
+			}
+		}
+	}

```

Comment:
Needs to be tested.

Reply:
In ASP.NET Core you can make the whole app require SSL, which would be okay and probably simpler, now that it's all using SSL anyway. This attribute is from the time when only certain pages were SSL. Even in development you could use SSL, because IIS Express supports it using self-signed certificate.